packages/medusa/src/services/cart.js

```diff
@@ -455,38 +439,24 @@ class CartService extends BaseService {
       if (currentItem) {
         const newQuantity = currentItem.quantity + lineItem.quantity
 
-        // Confirm inventory
-        const hasInventory = await this.confirmInventory_(
+        // Confirm inventory or throw error
+        await this.inventoryService_.confirmInventory(
```

Comment:
No reason to check inventory in both branches of the `if`-statement. Let us move it outside just before. And check that `item.variant_id` exists too. It might be a custom item

Reply:
And add `.withTransaction(manager)`